### PR TITLE
Francesco

### PR DESCRIPTION
**Pull request by Frankccv**

### CODE REVIEW

File: rules/96.5g_http2_dos_recognition.xml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+	counter ++;
+	if(counter >= limit_5g_dos_http2_initialuemessage_per_ms)
+		printf("Rule 96 Dos recognized\n");
+	return (counter >= limit_5g_dos_http2_initialuemessage_per_ms);
+}
+]]></embedded_functions>
+<property value="THEN" delay_units="ms" delay_min="0" delay_max="100" property_id="96" type_property="ATTACK" 
+    description="5G Http2 DoS attack Recognition">
+    <event value="COMPUTE" event_id="1" 
+        description="HTTP2 InitialUEMessage"
```
Comment (nhnghia) [inline rules/96.5g_http2_dos_recognition.xml]: Can you update this description?

File: rules/96.5g_http2_dos_recognition.xml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+<beginning>
+
```
Comment (nhnghia) [inline rules/96.5g_http2_dos_recognition.xml:2]: Can you add a description of this rule?

File: rules/96.5g_http2_dos_recognition.xml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+#include <stdlib.h>
+#include <mmt_core.h>
+
+
+#include<math.h>
+
+static unsigned long int limit_5g_dos_http2_initialuemessage_per_ms = 0;
+
+void on_load(){
+	const char *str = getenv("MMT_SEC_5G_DOS_NGAP_INITIALUEMESSAGE_MS_LIMIT");
```
Comment (nhnghia) [inline rules/96.5g_http2_dos_recognition.xml]: You need to change this environment variable's name as `NGAP_INITIALUEMESSAGE` is not applicable here

File: rules/96.5g_http2_dos_recognition.xml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+	if( str == NULL ){
+		mmt_info("Rule 96: no value of MMT_SEC_5G_DOS_NGAP_INITIALUEMESSAGE_MS_LIMIT" );
```
Comment (nhnghia) [inline rules/96.5g_http2_dos_recognition.xml]: `NGAP_INITIALUEMESSAGE` is not applicable

File: rules/96.5g_http2_dos_recognition.xml [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+		limit_5g_dos_http2_initialuemessage_per_ms=80;
+		
+	}
+	else
+		limit_5g_dos_http2_initialuemessage_per_ms = strtoul( str, NULL, 0 );
+	mmt_info("Rule 96: set MMT_SEC_5G_DOS_NGAP_INITIALUEMESSAGE_MS_LIMIT=%ld",
```
Comment (nhnghia) [inline rules/96.5g_http2_dos_recognition.xml]: `NGAP_INITIALUEMESSAGE` is not applicable

File: rules/96.5g_http2_dos_recognition.xml [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+		
+	}
+	else
+		limit_5g_dos_http2_initialuemessage_per_ms = strtoul( str, NULL, 0 );
+	mmt_info("Rule 96: set MMT_SEC_5G_DOS_NGAP_INITIALUEMESSAGE_MS_LIMIT=%ld", 
+		limit_5g_dos_http2_initialuemessage_per_ms );
+}
+
+/*
+To detect DoS, we group the InitialUEMessage by windows of millissecond,
```
Comment (nhnghia) [inline rules/96.5g_http2_dos_recognition.xml]: `InitialUEMessage` is not applicable

File: rules/97.http2_compression_attack_recognition.xml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+<beginning>
```
Comment (nhnghia) [inline rules/97.http2_compression_attack_recognition.xml:1]: Can you add a short description of this rule?

File: rules/98.http2_payload_fuzzing.xml [hunk 8]
```diff
@@ -0,0 +1,53 @@
+<beginning>
```
Comment (nhnghia) [inline rules/98.http2_payload_fuzzing.xml:1]: Can you add a short description of this rule?

File: rules/98.http2_payload_fuzzing.xml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+
+   	bool result=false;
+
+   	
+   	for (int i = 9; i < payload_length+9; i++) {
+   		//printf("  %02hhX ",payload[i]);
+    		  if((payload[i]=='\\')|| payload[i]=='#' || (payload[i]=='%')){
+    		  	result=true;
+    		  	 printf( " Rule 98:Recognized suspect fuzzing. Length %d  %f\n",payload_length,packet_id);
```
Comment (nhnghia) [inline rules/98.http2_payload_fuzzing.xml]: Need to remove this printed message

File: rules/98.http2_payload_fuzzing.xml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "mmt_lib.h"
+#include "tcpip/mmt_tcpip_protocols.h"
+#include "pre_embedded_functions.h"
+
+void on_load(){
+	printf("loading rule 98\n");
```
Comment (nhnghia) [inline rules/98.http2_payload_fuzzing.xml]: Need to remove this debug message

File: rules/97.http2_compression_attack_recognition.xml [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+#include <stdlib.h>
+#include "mmt_lib.h"
+#include "tcpip/mmt_tcpip_protocols.h"
+#include "pre_embedded_functions.h"
+
+
+static int em_check(
+	   double header_length,double packet_id){	
+
+   	printf( " Rule 97:Recognized suspect packet. Length %f , packet id %d\n",header_length,(int)packet_id);
```
Comment (nhnghia) [inline rules/97.http2_compression_attack_recognition.xml]: This debug message should be removed

File: rules/96.5g_http2_dos_recognition.xml [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+-->
+<embedded_functions><![CDATA[
+#include <stdlib.h>
+#include <mmt_core.h>
+
+
+#include<math.h>
+
+static unsigned long int limit_5g_dos_http2_initialuemessage_per_ms = 0;
```
Comment (nhnghia) [inline rules/96.5g_http2_dos_recognition.xml]: `initialuemessage` is not applicable here

File: rules/97.http2_compression_attack_recognition.xml (hunk 7 above)
Comment (Frankccv) [reply rules/97.http2_compression_attack_recognition.xml:1]: Sure, I added it

File: rules/96.5g_http2_dos_recognition.xml (hunk 4 above)
Comment (Frankccv) [reply rules/96.5g_http2_dos_recognition.xml]: I updated with the correct message

File: rules/96.5g_http2_dos_recognition.xml (hunk 1 above)
Comment (Frankccv) [reply rules/96.5g_http2_dos_recognition.xml]: Sure, I added a comment with the description of the rule

File: rules/96.5g_http2_dos_recognition.xml (hunk 6 above)
Comment (Frankccv) [reply rules/96.5g_http2_dos_recognition.xml]: I Deleted this comment

File: rules/98.http2_payload_fuzzing.xml (hunk 9 above)
Comment (Frankccv) [reply rules/98.http2_payload_fuzzing.xml]: Done!

File: rules/97.http2_compression_attack_recognition.xml (hunk 11 above)
Comment (Frankccv) [reply rules/97.http2_compression_attack_recognition.xml]: Message removed

File: rules/96.5g_http2_dos_recognition.xml (hunk 12 above)
Comment (Frankccv) [reply rules/96.5g_http2_dos_recognition.xml]: Name of threshold updated

**Review by nhnghia — APPROVED** [review]

**Pull request merged**